src/nxt_cgroup.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) Andrew Clayton
+ * Copyright (C) F5, Inc.
+ */
+
+#include <nxt_main.h>
+#include <nxt_process.h>
+#include <nxt_fs.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+
+
+static nxt_int_t nxt_make_cgpaths(const nxt_process_t *process, char *cgpath,
+    size_t cgpath_len, char *cgroot, size_t cgroot_len);
+static nxt_int_t nxt_cgroup_create(const nxt_task_t *task,
+    nxt_process_t *process);
+nxt_int_t nxt_cgroup_proc_add(const nxt_task_t *task, nxt_process_t *process);
+void nxt_cgroup_cleanup(const nxt_process_t *process);
+
+
+static nxt_int_t
+nxt_make_cgpaths(const nxt_process_t *process, char *cgpath, size_t cgpath_len,
+    char *cgroot, size_t cgroot_len)
+{
+    snprintf(cgpath, cgpath_len, NXT_CGROUP_ROOT "/%s",
+             process->isolation.cgroup.name);
+    if (cgroot != NULL) {
+        snprintf(cgroot, cgroot_len, "%s", NXT_CGROUP_ROOT);
+    }
+
+    return NXT_OK;
+}
+
+
+static nxt_int_t
+nxt_cgroup_create(const nxt_task_t *task, nxt_process_t *process)
+{
+    char    cgroot[PATH_MAX];
```

Comment:
Style:

Use only 2 spaces between type and var name.  See <https://nginx.org/en/docs/dev/development_guide.html#code_style> (we apply a slight variation regarding pointers, but you'll learn it from existing code when you need it).

Reply:
OK.